pkg/controller/dnszone/zonereconciler.go

```diff
@@ -107,30 +130,118 @@ func (zr *ZoneReconciler) Reconcile() error {
 		if err != nil {
 			zr.logger.WithError(err).Error("Failed to add finalizer to DNSZone")
 		}
-		return err
+		return reconcile.Result{}, err
 	}
 	if hostedZone == nil {
 		zr.logger.Debug("No corresponding hosted zone found on cloud provider, creating one")
 		hostedZone, err = zr.createRoute53HostedZone()
 		if err != nil {
 			zr.logger.WithError(err).Error("Failed to create hosted zone")
-			return err
+			return reconcile.Result{}, err
 		}
 	} else {
 		zr.logger.Debug("Existing hosted zone found. Syncing with DNSZone resource")
 		// For now, tags are the only things we can sync with existing zones.
 		err = zr.syncTags(hostedZone.Id, tags)
 		if err != nil {
 			zr.logger.WithError(err).Error("failed to sync tags for hosted zone")
-			return err
+			return reconcile.Result{}, err
 		}
 	}
+
 	nameServers, err := zr.getHostedZoneNSRecord(*hostedZone.Id)
 	if err != nil {
 		zr.logger.WithError(err).Error("Failed to get hosted zone name servers")
+		return reconcile.Result{}, err
+	}
+
+	if zr.dnsZone.Spec.LinkToParentDomain {
+		err = zr.syncParentDomainLink(nameServers)
+		if err != nil {
+			zr.logger.WithError(err).Error("failed syncing parent domain link")
+			return reconcile.Result{}, err
+		}
+	}
+
+	isZoneAvailable, err := zr.soaLookup(zr.dnsZone.Spec.Zone, zr.logger)
```

Comment:
Consider renaming to `isZoneSOAAvailable` as we already have a `isZoneAvailable` in the status.